src/boundary_conditions.jl

```diff
@@ -188,16 +156,56 @@ Return a default set of model boundary conditions. For now, this corresponds to
 doubly periodic domain, so `Periodic` boundary conditions along the x- and y-dimensions,
 with no-flux boundary conditions at the top and bottom.
 """
-function ModelBoundaryConditions()
-    bcs = (DoublyPeriodicBCs() for i = 1:length(solution_fields))
-    return ModelBoundaryConditions(bcs...)
+function ModelBoundaryConditions(;
+    u = DoublyPeriodicBCs(),
+    v = DoublyPeriodicBCs(),
+    w = DoublyPeriodicBCs(),
+    T = DoublyPeriodicBCs(),
+    S = DoublyPeriodicBCs()
+   )
+    return ModelBoundaryConditions(u, v, w, T, S)
+end
+
+function ChannelModelBoundaryConditions(;
+    u = ChannelBCs(),
+    v = ChannelBCs(),
+    w = ChannelBCs(),
+    T = ChannelBCs(),
+    S = ChannelBCs()
+   )
+    return ModelBoundaryConditions(u, v, w, T, S)
 end
 
-function ChannelModelBoundaryConditions()
-    bcs = (ChannelBCs() for i = 1:length(solution_fields))
-    return ModelBoundaryConditions(bcs...)
+#
+# Some helper functions for constructing boundary conditions
+#
+
+"""
+    BoundaryConditions(u=FieldBoundaryConditions(), ...)
+
+    BoundaryConditions(fld, coord, side, bc)
+
+Return an instance of `ModelBoundaryConditions` with one non-default
```

Comment:
Might just remove mentions of default and non-default which might be confusing as we don't use them anymore.